uxarray/grid/validation.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+from warnings import warn
+
+from uxarray.constants import ERROR_TOLERANCE
+
+
+# validation helper functions
+def _check_connectivity(self):
+    """Check if all nodes are referenced by at least one element.
+
+    If not, the mesh may have hanging nodes and may not a valid UGRID
+    mesh
+    """
+
+    # Check if all nodes are referenced by at least one element
+    # get unique nodes in connectivity
+    nodes_in_conn = np.unique(self.face_node_connectivity.values.flatten())
+    #  remove negative indices/fill values from the list
+    nodes_in_conn = nodes_in_conn[nodes_in_conn >= 0]
+
+    # check if the size of unique nodes in connectivity is equal to the number of nodes
+    if (nodes_in_conn.size == self.n_node):
+        print("-All nodes are referenced by at least one element.")
+        return True
+    else:
+        warn(
+            "Some nodes may not be referenced by any element. {0} and {1}".
+            format(nodes_in_conn.size, self.n_node), RuntimeWarning)
+        return False
+
+
+def _check_duplicate_nodes(self):
+    """Check if there are duplicate nodes in the mesh."""
+
+    coords1 = np.column_stack((np.vstack(self.node_x), np.vstack(self.node_y)))
```

Comment:
This should be `self.node_lon` and `self.node_lat` if you are intending to use Spherical coordinates (ignore if you meant Cartesian, but if so you should also check `node_z`